modules/router/src/services/config.ts

```diff
@@ -81,6 +97,157 @@ export const getSwapFees = (
   });
 };
 
+export const getMappedAssets = (_assetId: string, _chainId: number): { assetId: string; chainId: number }[] => {
+  const routerAllowedSwaps = getConfig().allowedSwaps;
+
+  let pair: { assetId: string; chainId: number }[] = [{ assetId: _assetId, chainId: _chainId }];
+  routerAllowedSwaps.map((s) => {
+    if (s.fromAssetId.toLowerCase() === _assetId.toLowerCase() && s.fromChainId === _chainId) {
+      pair.push({ assetId: getAddress(s.toAssetId), chainId: s.toChainId });
+    } else if (s.toAssetId.toLowerCase() === _assetId.toLowerCase() && s.toChainId === _chainId) {
+      pair.push({ assetId: getAddress(s.fromAssetId), chainId: s.fromChainId });
+    }
+  });
+
+  const uniquePairs = Array.from(new Set(pair.map((s) => s.assetId))).map((x) => {
+    return pair.find((z) => z.assetId === x)!;
+  });
+  return uniquePairs;
+};
+
+export const getPriceImpact = (marketPrice: BigNumber, estimatedPrice: BigNumber): BigNumber => {
```

Comment:
Do we return price impact as a percentage integer, as opposed to a decimal? Seems a bit dangerous - we are assuming method callers/users will know it's returning a perc integer.

I'm adding a method header just to clarify for now. We should be doing better docstrings soon, so this might not be worth concern rn

Reply:
Also, I thought price impact was only for effect of a token swap? Not diff between estimated price and market price in perc

Reply:
Math should be:
```
// Assuming user is swapping token B for (-->) token A in router:
// Get token's current prices.
priceTokenA = ..
priceTokenB = ..
// Get amounts for each token after executing the desired swap.
amountTokenA = <amount of token A in liquidity pool> - <amount being withdrawn>
amountTokenB = <amount of token B in liquidity pool> + <amount being deposited>
// Calculate new projected price of token after execution.
projectedPriceTokenB = amountTokenA * priceTokenA / amountTokenB
// Convert to percentage difference from previous value.
1 - projectedPriceTokenB / priceTokenB
```

Reply:
Alternatively, I could be misunderstanding this method's purpose, and it could just be a case of a misnomer here

Reply:
If you wanted to account for fees, assuming we take those in tokenA, we would change this one line:
```
// Assuming a 3% fee:
projectedPriceTokenB = 0.97 * amountTokenA * priceTokenA / amountTokenB
```